.ci/no-exception-test.sh

```diff
@@ -5,16 +5,30 @@ removeFolderWithProtectedFiles() {
   find $1 -delete
 }
 
+function checkout_from {
+  CLONE_URL=$1
+  PROJECT=$(echo "$CLONE_URL" | sed -nE 's/.*\/(.*).git/\1/p')
+  mkdir -p .ci-temp
+  cd .ci-temp
+  if [ -d "$PROJECT" ]; then
+    echo "Target project $PROJECT is already cloned, latest changes will be fetched"
+    cd $PROJECT
+    git fetch
+    cd ../
+  else
+    for i in 1 2 3 4 5; do git clone $CLONE_URL && break || sleep 15; done
+  fi
+  cd ../
+}
+
 case $1 in
 
 guava-with-google-checks)
   CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
   echo CS_version: $CS_POM_VERSION
-  mkdir -p .ci-temp/
-  cd .ci-temp/
-  git clone https://github.com/checkstyle/contribution
-  cd contribution/checkstyle-tester
+  checkout_from "-b issue-529 https://github.com/nmancus1/contribution.git"
```

Comment:
These changes seem unrelated to switching from launch to diff. I would move them to a new PR for faster merging as the swap from launch to diff will take some time.

Reply:
I am just now see your repo, so now I understand why the `-b` was added. My bad.

The addition of `checkout_from` seems beneficial, so it makes sense to me to have that in another PR.

Reply:
Done at https://github.com/checkstyle/checkstyle/pull/9306